libraries/common/stream.effekt

```diff
@@ -30,6 +30,16 @@ effect stop(): Nothing
 def for[A] { stream: () => Unit / emit[A] } { action: A => Unit }: Unit =
   returning::for[A, Unit]{stream}{action}
 
+/// Applies the given function on each emitted element and reemits the result.
+///
+/// e.g. map[Int,String]{ x => x.show } { range(0, 5) }
+def map[A, B] { function: A => B } { stream: () => Unit / emit[A] }: Unit / emit[B] =
+  returning::map[A, B, Unit]{function}{stream}
+
+/// Runs the given action whenever the given reader reads an element.
+def tap[A] { action: () => A / stop } { reader: () => Unit / read[A] }: Unit =
+  returning::tap[A, Unit]{action}{reader}
```

Comment:
Is there a nice example use-case for this function? I cannot think of a way to use it for something meaningful right now...
If so, could we add it to the description? 🙏